experimental/dds/tree2/src/test/shared-tree/fuzz/fuzzEditReducers.ts

```diff
@@ -20,53 +20,41 @@ import {
 	Operation,
 } from "./operationTypes";
 
-export const fuzzReducer = withNumberOfEditsCounted(
-	combineReducersAsync<Operation, FuzzTestState>({
-		edit: async (state, operation) => {
-			const { contents } = operation;
-			switch (contents.type) {
-				case "fieldEdit": {
-					const index = operation.index;
-					const tree = state.trees[index];
-					applyFieldEdit(tree, contents);
-					break;
-				}
-				case "nodeEdit": {
-					const change = operation.contents as NodeEdit;
-					const index = operation.index;
-					const tree = state.trees[index];
-					applyNodeEdit(tree, change.edit);
-					break;
-				}
-				default:
-					break;
+export const fuzzReducer = combineReducersAsync<
+	Operation | ChangeConnectionState,
+	DDSFuzzTestState<SharedTreeFactory>
+>({
+	edit: async (state, operation) => {
+		const { contents } = operation;
+		switch (contents.type) {
+			case "fieldEdit": {
+				const index = operation.index;
+				const tree = state.clients[index].channel;
+				applyFieldEdit(tree, contents);
+				break;
 			}
-			return state;
-		},
-		synchronize: async (state) => {
-			const { testTreeProvider } = state;
-			assert(testTreeProvider !== undefined);
-			await testTreeProvider.ensureSynchronized();
-			checkTreesAreSynchronized(state.trees);
-			return state;
-		},
-		transaction: async (state, operation) => {
-			const { contents, treeIndex } = operation;
-			const tree = state.trees[treeIndex];
-			applyTransactionEdit(tree, contents);
-			return state;
-		},
-	}),
-);
-
-function withNumberOfEditsCounted(
-	reducer: AsyncReducer<Operation, FuzzTestState>,
-): AsyncReducer<Operation, FuzzTestState> {
-	return async (state, op) => {
-		state.numberOfEdits++;
-		return reducer(state, op);
-	};
-}
+			case "nodeEdit": {
+				const change = operation.contents as NodeEdit;
+				const index = operation.index;
+				const tree = state.clients[index].channel;
+				applyNodeEdit(tree, change.edit);
+				break;
+			}
+			default:
+				break;
+		}
+		return state;
+	},
+	transaction: async (state, operation) => {
+		const { contents, treeIndex } = operation;
+		const tree = state.clients[treeIndex].channel;
+		applyTransactionEdit(tree, contents);
+		return state;
+	},
+	changeConnectionState: async (state, operation) => {
```

Comment:
you can also remove explicit `changeConnectionState` handling from your model. the harness handles it as a cross-cutting concern.

Reply:
removed `changeConnectionState`